src/transport/net.cu

```diff
@@ -183,6 +186,9 @@ ncclResult_t netGetRings(int nranks, int* groups, int* subgroups, ncclTvalue_t*
       }
       if (starts[group] == -1 || ends[group] == -1) {
         *nringsRet = ring;
+        free(starts);
```

Comment:
Once all alloc/free are outside the loop, we can call `goto final;` and have a `final:` label before the 3 calls to `free();` and `return ncclSuccess;`

Reply:
Done, here and in p2p.cu.